tests/Microsoft.Bot.Builder.AI.QnA.Tests/QnAMakerTests.cs

```diff
@@ -1138,6 +1138,51 @@ public async Task QnaMaker_Test_Options_Hydration()
             Assert.AreEqual(0, requestContent6.StrictFilters.Length);
         }
 
+        [TestMethod]
+        [TestCategory("AI")]
+        [TestCategory("QnAMaker")]
+        public async Task QnaMaker_StrictFilters_Compound_OperationType()
+        {
+            var mockHttp = new MockHttpMessageHandler();
+            mockHttp.When(HttpMethod.Post, GetRequestUrl())
+                .Respond("application/json", GetResponse("QnaMaker_ReturnsAnswer.json"));
+
+            var interceptHttp = new InterceptRequestHandler(mockHttp);
+            var oneFilteredOption = new QnAMakerOptions()
+            {
+                Top = 30,
+                StrictFilters = new Metadata[]
+                {
+                    new Metadata()
+                    {
+                        Name = "movie",
+                        Value = "disney",
+                    },
+                    new Metadata()
+                    {
+                        Name = "production",
+                        Value = "Walden",
+                    },
+                },
+                StrictFiltersCompoundOperationType = StrictFiltersCompoundOperationType.OR
+            };
+            var qna = GetQnAMaker(
+                            interceptHttp,
+                            new QnAMakerEndpoint
+                            {
+                                KnowledgeBaseId = _knowledgeBaseId,
+                                EndpointKey = _endpointKey,
+
+                                Host = _hostname,
+                            }, oneFilteredOption);
+            
+            var context = GetContext("up");
+            var noFilterResults1 = await qna.GetAnswersAsync(context, oneFilteredOption);
+            var requestContent1 = JsonConvert.DeserializeObject<CapturedRequest>(interceptHttp.Content);
+
+            Assert.AreEqual(1, noFilterResults1.Length);
```

Comment:
here we need to verify , number of string fliters and is compound operation type set to expected value or not right?

Reply:
Implemented review comments.

---
In reply to: [470579737](https://github.com/microsoft/botbuilder-dotnet/pull/4446#discussion_r470579737) [](ancestors = 470579737)